.github/workflows/java.yml

```
@@ -0,0 +1,74 @@
+jobs:
+  # Job for cleaning, building, and testing the project
+  build_and_test:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the code from the repository
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # Set up Java 21 with Maven
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
```

actionlint (1.7.4)

the runner of "actions/setup-java@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)